packages/app/src/cli/services/app_logs/write-app-logs.test.ts

```diff
@@ -0,0 +1,63 @@
+import {writeAppLogsToFile} from './write-app-logs.js'
+import {AppEventData} from './fetch-app-logs.js'
+import {joinPath} from '@shopify/cli-kit/node/path'
+import {writeLog} from '@shopify/cli-kit/node/logs'
+import {describe, expect, test, vi, beforeEach} from 'vitest'
+
+vi.mock('@shopify/cli-kit/node/logs')
+
+const APP_LOG: AppEventData = {
+  shop_id: 1,
+  api_client_id: 2,
+  payload:
+    '{"input":"{\\"cart\\":{\\"lines\\":[{\\"quantity\\":1,\\"merchandise\\":{\\"__typename\\":\\"ProductVariant\\",\\"id\\":\\"gid:\\\\\\/\\\\\\/shopify\\\\\\/ProductVariant\\\\\\/2\\"}}]}}","input_bytes":123,"output":"{\\"discountApplicationStrategy\\":\\"FIRST\\",\\"discounts\\":[{\\"message\\":\\"10% off\\",\\"value\\":{\\"percentage\\":{\\"value\\":10}},\\"targets\\":[{\\"productVariant\\":{\\"id\\":\\"gid:\\/\\/shopify\\/ProductVariant\\/2\\"}}]}]}","output_bytes":182,"function_id":"e57b4d31-2038-49ff-a0a1-1eea532414f7","logs":"logging in the 10% discount\\ninput: \\n[object Object]\\n","fuel_consumed":512436}',
+  event_type: 'function_run',
+  cursor: '2024-05-22T15:06:43.841156Z',
+  status: 'success',
+  log_timestamp: '2024-05-22T15:06:41.827379Z',
+}
+const API_KEY = 'apiKey'
+
+describe('writeAppLogsToFile', () => {
+  let stdout: any
+
+  beforeEach(() => {
+    stdout = {write: vi.fn()}
+  })
+
+  test('calls writeLog with the right data', async () => {
+    // Given
+    const logData = expectedLogDataFromAppEvent(APP_LOG)
+
+    // determine the fileName and path
+    const fileName = `app_logs_${APP_LOG.log_timestamp}.json`
+    const path = joinPath(API_KEY, fileName)
+
+    // When
+    await writeAppLogsToFile({appLog: APP_LOG, apiKey: API_KEY, stdout})
```

Comment:
This will write to the "real" logs location (but for a fake app). Maybe a temporary folder would be better? There are good test utils to help with that used elsewhere, that will tear down after a test finishes.